packages/ipns/src/dns-resolvers/dns-over-https.ts

```diff
@@ -131,11 +126,19 @@ function toDNSResponse (response: dnsPacket.Packet): DNSResponse {
         }
       }
 
+      if (!Buffer.isBuffer(a.data[0])) {
+        return {
+          name: a.name,
+          type: txtType,
+          TTL: a.ttl ?? ttl,
+          data: String(a.data[0])
+        }
+      }
+
       return {
         name: a.name,
         type: txtType,
         TTL: a.ttl ?? ttl,
-        // @ts-expect-error we have already checked that a.data is not empty
         data: uint8ArrayToString(a.data[0])
```

Comment:
The TS error was not that data was empty, but that `a.data[0] === string | number | Buffer`

We may need more smarts around this check

Reply:
around the *above check of `!Buffer.isBuffer(a.data[0])`